R/air.R

```diff
@@ -0,0 +1,214 @@
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' # Prepare an R package or project to use Air
+#' use_air()
+#' }
+use_air <- function(vscode = TRUE) {
+  check_bool(vscode)
+
+  ignore <- is_package()
+
+  # Create empty `air.toml` if it doesn't exist
+  create_air_toml(ignore = ignore)
+
+  if (vscode) {
+    create_vscode_directory(ignore = ignore)
+
+    # Create project level `settings.json` if it doesn't exist,
+    # and write in Air specific formatter settings
+    path <- create_vscode_json_file("settings.json")
+    write_air_vscode_settings_json(path)
+
+    # Create project level `extensions.json` if it doesn't exist,
+    # and write in Air as a recommended extension for this project
+    path <- create_vscode_json_file("extensions.json")
+    write_air_vscode_extensions_json(path)
+  }
+
+  ui_bullets(c(
+    "_" = "Read the {.href [Air editors guide](https://posit-dev.github.io/air/editors.html)}
+           to learn how to invoke Air in your preferred editor."
+  ))
+
+  invisible(TRUE)
+}
+
+#' Creates an empty `air.toml`
+#'
+#' If either `air.toml` or `.air.toml` already exist, no new file is created.
+#'
+#' @keywords internal
+#' @noRd
+create_air_toml <- function(ignore = FALSE) {
+  path <- path_first_existing(proj_path(c("air.toml", ".air.toml")))
+
+  if (is.null(path)) {
+    # No pre-existing configuration file, create it
+    path <- proj_path("air.toml")
+    file_create(path)
+    ui_bullets(c("v" = "Creating {.path {pth(path)}}."))
+  }
+
+  if (ignore) {
+    use_build_ignore(air_toml_regex(), escape = FALSE)
+  }
+
+  invisible(path)
+}
+
+air_toml_regex <- function() {
+  # Pre-escaped regex allowing both `air.toml` and `.air.toml`
+  "^[\\.]?air\\.toml$"
+}
+
+create_vscode_json_file <- function(name) {
+  arg_match(name, values = c("settings.json", "extensions.json"))
+
+  path <- proj_path(".vscode", name)
+
+  if (!file_exists(path)) {
+    file_create(path)
+    ui_bullets(c("v" = "Creating {.path {pth(path)}}."))
+  }
+
+  # Tools like jsonlite fail to read empty json files,
+  # so if we've just created it, write in `{}`. The easiest
+  # way to do that is to write an empty named list.
+  if (is_file_empty(path)) {
+    jsonlite::write_json(set_names(list()), path = path, pretty = TRUE)
+  }
+
+  invisible(path)
+}
+
+write_air_vscode_settings_json <- function(path) {
+  settings <- jsonlite::read_json(path)
+  settings_r <- settings[["[r]"]]
+
+  if (is.null(settings_r)) {
+    # Mock it
+    settings_r <- set_names(list())
+  }
+
+  # Set these regardless of their previous values. Assume that calling
+  # `use_air()` is an explicit request to opt in to these settings.
+  settings_r[["editor.formatOnSave"]] <- TRUE
+  settings_r[["editor.defaultFormatter"]] <- "Posit.air-vscode"
+
+  settings[["[r]"]] <- settings_r
+
+  write_vscode_json(x = settings, path = path)
+}
+
+write_air_vscode_extensions_json <- function(path) {
+  settings <- jsonlite::read_json(path)
```

Comment:
I think this operation may fail because `settings.json` is actually JSONC (JSON with Comments), not JSON.